<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> publications | </title> <meta name="author" content="Shicong (Mimi) Xie"> <meta name="description" content="publications by categories in reversed chronological order. generated by jekyll-scholar."> <meta name="keywords" content="jekyll, jekyll-theme, academic-website, portfolio-website"> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link defer rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons&amp;display=swap"> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="data:image/svg+xml,&lt;svg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%20100%20100%22&gt;&lt;text%20y=%22.9em%22%20font-size=%2290%22&gt;%E2%9A%9B%EF%B8%8F&lt;/text&gt;&lt;/svg&gt;"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://xies.github.io/publications/"> <script src="/assets/js/theme.js?9a0c749ec5240d9cda97bc72359a72c0"></script> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script>initTheme();</script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"> </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">about </a> </li> <li class="nav-item "> <a class="nav-link" href="/blog/">blog </a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">cv </a> </li> <li class="nav-item active"> <a class="nav-link" href="/publications/">publications <span class="sr-only">(current)</span> </a> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="ti ti-sun-moon" id="light-toggle-system"></i> <i class="ti ti-moon-filled" id="light-toggle-dark"></i> <i class="ti ti-sun-filled" id="light-toggle-light"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title">publications</h1> <p class="post-description">publications by categories in reversed chronological order. generated by jekyll-scholar.</p> </header> <article> <div class="publications"> <h2 class="bibliography">2024</h2> <ol class="bibliography"><li> <div class="row"> <div class="col col-sm-2 abbr"> <abbr class="badge rounded w-100">bioRxiv</abbr> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/cell_autonomous.png" sizes="300px"></source> <img src="/assets/img/publication_preview/cell_autonomous.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="cell_autonomous.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="xie_g1s_2024" class="col-sm-8"> <div class="title">The G1/S transition in mammalian stem cells in vivo is autonomously regulated by cell size</div> <div class="author"> <em>Shicong Xie</em>, Shuyuan Zhang, Gustavo de Medeiros, Prisca Liberali, and <span class="more-authors" title="click to view 1 more author" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '1 more author' ? 'Jan M. Skotheim' : '1 more author'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">1 more author</span> </div> <div class="periodical"> Apr 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.biorxiv.org/content/10.1101/2024.04.09.588781v1" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="abstract hidden"> <p>Cell growth and division must be coordinated to maintain a stable cell size, but how this coordination is implemented in multicellular tissues remains unclear. In unicellular eukaryotes, autonomous cell size control mechanisms couple cell growth and division with little extracellular input. However, in multicellular tissues we do not know if autonomous cell size control mechanisms operate the same way or whether cell growth and cell cycle progression are separately controlled by cell-extrinsic signals. Here, we address this question by tracking single epidermal stem cells growing in adult mice. We find that a cell-autonomous size control mechanism, dependent on the RB pathway, sets the timing of S phase entry based on the cell’s current size. Cell-extrinsic variations in the cellular microenvironment affect cell growth rates but not this autonomous coupling. Our work reassesses long-standing models of cell cycle regulation within complex metazoan tissues and identifies cell-autonomous size control as a critical mechanism regulating cell divisions in vivo and thereby a major contributor to stem cell heterogeneity.</p> </div> </div> </div> </li></ol> <h2 class="bibliography">2022</h2> <ol class="bibliography"><li> <div class="row"> <div class="col col-sm-2 abbr"> <abbr class="badge rounded w-100">Ann. Rev. Cell Bio</abbr> </div> <div id="xie_eukaryotic_2022" class="col-sm-8"> <div class="title">Eukaryotic Cell Size Control and Its Relation to Biosynthesis and Senescence</div> <div class="author"> <em>Shicong Xie</em>, Matthew Swaffer, and Jan M. Skotheim </div> <div class="periodical"> <em>Annual Review of Cell and Developmental Biology</em>, Apr 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> </div> <div class="abstract hidden"> <p>The most fundamental feature of cellular form is size, which sets the scale of all cell biological processes. Growth, form, and function are all necessarily linked in cell biology, but we often do not understand the underlying molecular mechanisms nor their specific functions. Here, we review progress toward determining the molecular mechanisms that regulate cell size in yeast, animals, and plants, as well as progress toward understanding the function of cell size regulation. It has become increasingly clear that the mechanism of cell size regulation is deeply intertwined with basic mechanisms of biosynthesis, and how biosynthesis can be scaled (or not) in proportion to cell size. Finally, we highlight recent findings causally linking aberrant cell size regulation to cellular senescence and their implications for cancer therapies. Expected final online publication date for the Annual Review of Cell and Developmental Biology Volume 38 is October 2022. Please see http://www.annualreviews.org/page/journal/pubdates for revised estimates.</p> </div> </div> </div> </li></ol> <h2 class="bibliography">2021</h2> <ol class="bibliography"><li> <div class="row"> <div class="col col-sm-2 abbr"> <abbr class="badge rounded w-100">Curr. Biol.</abbr> </div> <div id="xie_cell-size_2021" class="col-sm-8"> <div class="title">Cell-size control: Chromatin-based titration primes inhibitor dilution</div> <div class="author"> <em>Shicong Xie</em>, and Jan M. Skotheim </div> <div class="periodical"> <em>Current Biology</em>, Oct 2021 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> </div> <div class="abstract hidden"> <p>Cell growth can drive progression into the cell cycle by diluting a diverse set of cell-cycle inhibitors in yeast, animal, and plant cells. Inhibitor dilution mechanisms implement cell-size control when large and small cells inherit a similar number of inhibitor molecules, and new work shows that these mechanisms in plant cells include specific degradation and chromatin-partitioning components.</p> </div> </div> </div> </li></ol> <h2 class="bibliography">2020</h2> <ol class="bibliography"><li> <div class="row"> <div class="col col-sm-2 abbr"> <abbr class="badge rounded w-100">Curr. Biol.</abbr> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/cell_size_homeostasis.png" sizes="300px"></source> <img src="/assets/img/publication_preview/cell_size_homeostasis.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="cell_size_homeostasis.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="xie_g1_2020" class="col-sm-8"> <div class="title">A G1 Sizer Coordinates Growth and Division in the Mouse Epidermis.</div> <div class="author"> <em>Shicong Xie</em>, and Jan M Skotheim </div> <div class="periodical"> <em>Current biology</em>, Oct 2020 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> </div> <div class="abstract hidden"> <p>Cell size homeostasis is often achieved by coupling cell-cycle progression to cell growth. Growth has been shown to drive cell-cycle progression in bacteria and yeast through "sizers," wherein cells of varying birth size divide at similar final sizes [1-3], and "adders," wherein cells increase in size a fixed amount per cell cycle [4-6]. Intermediate control phenomena are also observed, and even the same organism can exhibit different control phenomena depending on growth conditions [2, 7, 8]. Although studying unicellular organisms in laboratory conditions may give insight into their growth control in the wild, this is less apparent for studies of mammalian cells growing outside the organism. Sizers, adders, and intermediate phenomena have been observed in vitro [9-12], but it is unclear how this relates to mammalian cell proliferation in vivo. To address this question, we analyzed time-lapse images of the mouse epidermis taken over 1 week during normal tissue turnover [13]. We quantified the 3D volume growth and cell-cycle progression of single cells within the mouse skin. In dividing epidermal stem cells, we found that cell growth is coupled to division through a sizer operating largely in the G1 phase of the cell cycle. Thus, although the majority of tissue culture studies have identified adders, our analysis demonstrates that sizers are important in vivo and highlights the need to determine their underlying molecular origin.</p> </div> </div> </div> </li></ol> <h2 class="bibliography">2019</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col col-sm-2 abbr"> <abbr class="badge rounded w-100">Mol. Biol. Cell</abbr> </div> <div id="berenson_constitutive_2019" class="col-sm-8"> <div class="title">Constitutive expression of a fluorescent protein reports the size of live human cells.</div> <div class="author"> Daniel F Berenson, Evgeny Zatulovskiy, <em>Shicong Xie</em>, and Jan M Skotheim </div> <div class="periodical"> <em>Molecular biology of the cell</em>, Oct 2019 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.molbiolcell.org/doi/10.1091/mbc.E19-03-0171" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="abstract hidden"> <p>Cell size is important for cell physiology because it sets the geometric scale of organelles and biosynthesis. A number of methods exist to measure different aspects of cell size, but each has significant drawbacks. Here, we present an alternative method to measure the size of single human cells using a nuclear localized fluorescent protein expressed from a constitutive promoter. We validate this method by comparing it to several established cell size measurement strategies, including flow cytometry optical scatter, total protein dyes, and quantitative phase microscopy. We directly compare our fluorescent protein measurement with the commonly used measurement of nuclear volume and show that our measurements are more robust and less dependent on image segmentation. We apply our method to examine how cell size impacts the cell division cycle and reaffirm that there is a negative correlation between size at cell birth and G1 duration. Importantly, combining our size reporter with fluorescent labeling of a different protein in a different color channel allows measurement of concentration dynamics using simple wide-field fluorescence imaging. Thus, we expect our method will be of use to researchers interested in how dynamically changing protein concentrations control cell fates.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <abbr class="badge rounded w-100">Mol. Cell</abbr> </div> <div id="shariati_reversible_2019" class="col-sm-8"> <div class="title">Reversible Disruption of Specific Transcription Factor-DNA Interactions Using CRISPR/Cas9.</div> <div class="author"> S Ali Shariati, Antonia Dominguez, <em>Shicong Xie</em>, Marius Wernig, and <span class="more-authors" title="click to view 2 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '2 more authors' ? 'Lei S Qi, Jan M Skotheim' : '2 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">2 more authors</span> </div> <div class="periodical"> <em>Molecular cell</em>, Oct 2019 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> </div> <div class="abstract hidden"> <p>The control of gene expression by transcription factor binding sites frequently determines phenotype. However, it is difficult to determine the function of single transcription factor binding sites within larger transcription networks. Here, we use deactivated Cas9 (dCas9) to disrupt binding to specific sites, a method we term CRISPRd. Since CRISPR guide RNAs are longer than transcription factor binding sites, flanking sequence can be used to target specific sites. Targeting dCas9 to an Oct4 site in the Nanog promoter displaced Oct4 from this site, reduced Nanog expression, and slowed division. In contrast, disrupting the Oct4 binding site adjacent to Pax6 upregulated Pax6 transcription and disrupting Nanog binding its own promoter upregulated its transcription. Thus, we can easily distinguish between activating and repressing binding sites and examine autoregulation. Finally, multiple guide RNA expression allows simultaneous inhibition of multiple binding sites, and conditionally destabilized dCas9 allows rapid reversibility.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <abbr class="badge rounded w-100">Mol. Cell</abbr> </div> <div id="topacio_cyclin_2019" class="col-sm-8"> <div class="title">Cyclin D-Cdk4,6 Drives Cell-Cycle Progression via the Retinoblastoma Protein’s C-Terminal Helix.</div> <div class="author"> Benjamin R Topacio, Evgeny Zatulovskiy, Sandra Cristea, <em>Shicong Xie</em>, and <span class="more-authors" title="click to view 5 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '5 more authors' ? 'Carrie S Tambo, Seth M Rubin, Julien Sage, Mardo Kõivomägi, Jan M Skotheim' : '5 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">5 more authors</span> </div> <div class="periodical"> <em>Molecular cell</em>, Oct 2019 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> </div> <div class="abstract hidden"> <p>The cyclin-dependent kinases Cdk4 and Cdk6 form complexes with D-type cyclins to drive cell proliferation. A well-known target of cyclin D-Cdk4,6 is the retinoblastoma protein Rb, which inhibits cell-cycle progression until its inactivation by phosphorylation. However, the role of Rb phosphorylation by cyclin D-Cdk4,6 in cell-cycle progression is unclear because Rb can be phosphorylated by other cyclin-Cdks, and cyclin D-Cdk4,6 has other targets involved in cell division. Here, we show that cyclin D-Cdk4,6 docks one side of an alpha-helix in the Rb C terminus, which is not recognized by cyclins E, A, and B. This helix-based docking mechanism is shared by the p107 and p130 Rb-family members across metazoans. Mutation of the Rb C-terminal helix prevents its phosphorylation, promotes G1 arrest, and enhances Rb’s tumor suppressive function. Our work conclusively demonstrates that the cyclin D-Rb interaction drives cell division and expands the diversity of known cyclin-based protein docking mechanisms.</p> </div> </div> </div> </li> </ol> <h2 class="bibliography">2016</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col col-sm-2 abbr"> <abbr class="badge rounded w-100">J. Cell Biol.</abbr> </div> <div id="mason_rhoa_2016" class="col-sm-8"> <div class="title">RhoA GTPase inhibition organizes contraction during epithelial morphogenesis</div> <div class="author"> Frank M Mason, <em>Shicong Xie</em>, Claudia G Vasquez, Michael Tworoger, and <span class="more-authors" title="click to view 1 more author" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '1 more author' ? 'Adam C Martin' : '1 more author'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">1 more author</span> </div> <div class="periodical"> <em>The Journal of cell biology</em>, Oct 2016 </div> <div class="periodical"> </div> <div class="links"> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <abbr class="badge rounded w-100">Mol. Biol. Cell</abbr> </div> <div id="xie_loss_2016" class="col-sm-8"> <div class="title">Loss of Gα12/13 exacerbates apical area dependence of actomyosin contractility</div> <div class="author"> <em>Shicong Xie</em>, Frank M Mason, and Adam C Martin </div> <div class="periodical"> <em>Molecular Biology of the Cell</em>, Oct 2016 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> </div> <div class="abstract hidden"> <p>During development, coordinated cell shape changes alter tissue shape. In the Drosophila ventral furrow and other epithelia, apical constriction of hundreds of epithelial cells folds the tissue. Genes in the Gα12/13 pathway coordinate collective apical constriction, but the mechanism of coordination is poorly understood. Coupling live-cell imaging with a computational approach to identify contractile events, we discovered that differences in constriction behavior are biased by initial cell shape. Disrupting Gα12/13 exacerbates this relationship. Larger apical area is associated with delayed initiation of contractile pulses, lower apical E-cadherin and F-actin levels, and aberrantly mobile Rho-kinase structures. Our results suggest that loss of Gα12/13 disrupts apical actin cortex organization and pulse initiation in a sizedependent manner. We propose that Gα12/13 robustly organizes the apical cortex despite variation in apical area to ensure the timely initiation of contractile pulses in a tissue with heterogeneity in starting cell shape.</p> </div> </div> </div> </li> </ol> <h2 class="bibliography">2015</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col col-sm-2 abbr"> <abbr class="badge rounded w-100">Cell Rep.</abbr> </div> <div id="tikhonova_transcription_2015" class="col-sm-8"> <div class="title">A transcription blocker isolated from a designed repeat protein combinatorial library by in vivo functional screen</div> <div class="author"> Elena B Tikhonova, Abdul S Ethayathulla, Yue Su, Parameswaran Hariharan, and <span class="more-authors" title="click to view 2 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '2 more authors' ? 'Shicong Xie, Lan Guan' : '2 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">2 more authors</span> </div> <div class="periodical"> <em>Scientific reports</em>, Oct 2015 </div> <div class="periodical"> </div> <div class="links"> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <abbr class="badge rounded w-100">Nat. Comm.</abbr> </div> <div id="xie_intracellular_2015" class="col-sm-8"> <div class="title">Intracellular signalling and intercellular coupling coordinate heterogeneous contractile events to facilitate tissue folding.</div> <div class="author"> <em>Shicong Xie</em>, and Adam C Martin </div> <div class="periodical"> <em>Nature communications</em>, Oct 2015 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> </div> <div class="abstract hidden"> <p>Cellular forces generated in the apical domain of epithelial cells reshape tissues. Recent studies highlighted an important role for dynamic actomyosin contractions, called pulses, that change cell and tissue shape. Net cell shape change depends on whether cell shape is stabilized, or ratcheted, between pulses. Whether there are different classes of contractile pulses in wild-type embryos and how pulses are spatiotemporally coordinated is unknown. Here we develop a computational framework to identify and classify pulses and determine how pulses are coordinated during invagination of the Drosophila ventral furrow. We demonstrate biased transitions in pulse behaviour, where weak or unratcheted pulses transition to ratcheted pulses. The transcription factor Twist directs this transition, with cells in Twist-depleted embryos exhibiting abnormal reversed transitions in pulse behaviour. We demonstrate that ratcheted pulses have higher probability of having neighbouring contractions, and that ratcheting of pulses prevents competition between neighbouring contractions, allowing collective behaviour.</p> </div> </div> </div> </li> </ol> </div> </article> </div> </div> <footer class="fixed-bottom" role="contentinfo"> <div class="container mt-0"> © Copyright 2024 Shicong (Mimi) Xie. Last updated: September 20, 2024. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@5.0.0/imagesloaded.pkgd.min.js" integrity="sha256-htrLFfZJ6v5udOG+3kNLINIKh2gvoKqwEhHYfTTMICc=" crossorigin="anonymous"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.1.0/dist/medium-zoom.min.js" integrity="sha256-ZgMyDAIYDYGxbcpJcfUnYwNevG/xi9OHKaR/8GK+jWc=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script src="/assets/js/no_defer.js?2781658a0a2b13ed609542042a859126"></script> <script defer src="/assets/js/common.js?e0514a05c5c95ac1a93a8dfd5249b92e"></script> <script defer src="/assets/js/copy_code.js?12775fdf7f95e901d7119054556e495f" type="text/javascript"></script> <script defer src="/assets/js/jupyter_new_tab.js?d9f17b6adc2311cbabd747f4538bb15f"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script> <script defer src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6" crossorigin="anonymous"></script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> <script src="/assets/js/vanilla-back-to-top.min.js?f40d453793ff4f64e238e420181a1d17"></script> <script>addBackToTop();</script> </body> </html>